ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java

```diff
@@ -306,23 +306,32 @@ public String toString() {
   private final boolean skipPreVote;
   private final ConfAndTerm round0;
 
-  LeaderElection(RaftServerImpl server, boolean force) {
-    this(ServerInterface.get(server), force);
+  static LeaderElection newInstance(RaftServerImpl server, boolean force) {
+    return newInstance(ServerInterface.get(server), force);
   }
 
-  LeaderElection(ServerInterface server, boolean force) {
-    this.name = ServerStringUtils.generateUnifiedName(server.getMemberId(), getClass()) + COUNT.incrementAndGet();
+  static LeaderElection newInstance(ServerInterface server, boolean force) {
+      String name = "";
+      try {
+        // increase term of the candidate in advance if it's forced to election
+        final ConfAndTerm round0 = force ? server.initElection(Phase.ELECTION) : null;
+        name = ServerStringUtils.generateUnifiedName(server.getMemberId(), LeaderElection.class)
+              + COUNT.incrementAndGet();
```

Comment:
- `name` should be set before `initElection`, otherwise it will still be `""` in `catch`.  I think it can be set even before `try`.
- nit: indentation should be less by 2 spaces

Reply:
Nice catch! I have just made a mistake...

Has Fixed